ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/ontologies/model/OntologyHierarchyTreeStoreProvider.java

```diff
@@ -0,0 +1,62 @@
+package org.iplantc.de.admin.desktop.client.ontologies.model;
+
+import org.iplantc.de.client.models.ontologies.OntologyHierarchy;
+
+import com.google.common.base.Strings;
+import com.google.inject.Provider;
+import com.google.web.bindery.autobean.shared.AutoBean;
+import com.google.web.bindery.autobean.shared.AutoBeanUtils;
+
+import com.sencha.gxt.data.shared.ModelKeyProvider;
+import com.sencha.gxt.data.shared.TreeStore;
+
+/**
+ * @author aramsey
+ */
+public class OntologyHierarchyTreeStoreProvider implements Provider<TreeStore<OntologyHierarchy>> {
+
+    private static final String HIERARCHY_PARENT_MODEL_KEY = "parent_key";
+    private static final String HIERARCHY_MODEL_KEY = "model_key";
+
+    @Override
+    public TreeStore<OntologyHierarchy> get() {
+        return new TreeStore<>(new ModelKeyProvider<OntologyHierarchy>() {
+            @Override
+            public String getKey(OntologyHierarchy item) {
+                String key = getHierarchyPathTag(item);
+                setChildrenParentTag(key, item);
+                return key;
```

Comment:
Since IRI's aren't actually unique in a hierarchy (for example, something vague like `Analysis` with an IRI of topic_1234 could exist under `Biology` and `Computer Science`), the key for the TreeStore is the path to that specific hierarchy.
The path/key is stored in a tag. Each `OntologyHierarchy` has two tags - one for its parent (which is the parent's full path), and one for itself (which is the parent's path + its own label).